### PR TITLE
samples: bluetooth: Update distance estimation for mode 3 and multi a…

### PR DESCRIPTION
**Pull request by sean-madigan**
…ntenna

Mode 3 is a combination of modes 1 and 2, so extract out some of the common logic and use this for mode 3.

Adjust array sizes and indexes so they can be used for multi antenna.

### CODE REVIEW

File: samples/bluetooth/channel_sounding_ras_initiator/src/distance_estimation.c [hunk 1]
```diff
@@ -186,68 +257,31 @@ static bool process_step_data(struct bt_le_cs_subevent_step *local_step,
...
-				local_step_data->toa_tod_initiator;
-			mode_1_data[context->mode_1_data_index].tod_toa_reflector =
-				peer_step_data->tod_toa_reflector;
-		} else if (context->role == BT_CONN_LE_CS_ROLE_REFLECTOR) {
-			mode_1_data[context->mode_1_data_index].tod_toa_reflector =
-				local_step_data->tod_toa_reflector;
-			mode_1_data[context->mode_1_data_index].toa_tod_initiator =
-				peer_step_data->toa_tod_initiator;
-		}
+	} else if (local_step->mode == BT_HCI_OP_LE_CS_MAIN_MODE_3) {
```
Comment (olivier-le-sage) [inline samples/bluetooth/channel_sounding_ras_initiator/src/distance_estimation.c:272]: would probably be a bit more logical to add this later (when the sample actually generates mode 3 steps)

**Review by olivier-le-sage — APPROVED** [review]
would be good to run the sample test locally as since it doesn't run in babblesim/oncommit

File: samples/bluetooth/channel_sounding_ras_initiator/src/distance_estimation.c (hunk 1 above)
Comment (sean-madigan) [reply samples/bluetooth/channel_sounding_ras_initiator/src/distance_estimation.c:272]: don't see a down side to merging this now

File: samples/bluetooth/channel_sounding_ras_initiator/src/distance_estimation.c [hunk 2]
```diff
@@ -21,7 +21,8 @@ LOG_MODULE_DECLARE(app_main, LOG_LEVEL_INF);
 #define SPEED_OF_LIGHT_M_PER_S	(299792458.0f)
 #define SPEED_OF_LIGHT_NM_PER_S (SPEED_OF_LIGHT_M_PER_S / 1000000000.0f)
 #define PI			3.14159265358979323846f
-#define MAX_NUM_SAMPLES		256
+#define MAX_NUM_RTT_SAMPLES		256
+#define MAX_NUM_IQ_SAMPLES		256 * CONFIG_BT_RAS_MAX_ANTENNA_PATHS
```
Comment (olivier-le-sage) [inline samples/bluetooth/channel_sounding_ras_initiator/src/distance_estimation.c:25]: it's not really related to this PR but I suppose another improvement would be to process the tone extension slots

**Review by thomas-m-johansen — APPROVED** [review]
lgtm

**Review by weeTike — APPROVED** [review]

**Pull request merged**